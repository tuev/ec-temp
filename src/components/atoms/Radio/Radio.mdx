import { Meta, Story, Preview } from '@storybook/addon-docs/blocks'
import * as stories from './Radio.stories.tsx'
import Radio from './Radio.tsx'

<Story name="basic" />
<Story name="outer" />
