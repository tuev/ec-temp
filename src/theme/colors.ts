export interface IPalette {
  light: string
  main: string
  dark: string
  contrastText: string
}

export type IColorPalette = {
  [key in colorType]: IPalette
}

export enum COLOR {
  'primary' = '#F62F5E',
  'black' = '#2E2E2E',
  'white' = '#fff',
  'blue' = '#6EB2FB',
  'orange' = '#F1AD3D',
  'gray' = '#F7F7F7',
  'green' = '#00D3CA',
  'yellow' = '##EFFC90',
  'black1' = '#6C6C6C',
  'black2' = '#B4B4B4',
}

export type colorType =
  | 'primary'
  | 'blue'
  | 'black'
  | 'orange'
  | 'gray'
  | 'green'
  | 'yellow'
  | 'red'

export const redPalette = {
  contrastText: '#FFF',
  dark: '#cf2557',
  light: '#fc688e',
  main: '#F62F5E',
}

export const primaryPalette = {
  contrastText: '#FFF',
  dark: '#cf2557',
  light: '#fc688e',
  main: '#F62F5E',
}

export const bluePalette = {
  contrastText: '#FFF',
  dark: '#4592f7',
  light: '#beddfe',
  main: '#6EB2FB',
}

export const blackPalette = {
  contrastText: '#FFF',
  dark: '#2e2e2e',
  light: '#b0b0b0',
  main: '#707070',
}

export const orangePalette = {
  contrastText: '#FFF',
  dark: '#ea862e',
  light: '#f8da4f',
  main: '#F1AD3D',
}
export const grayPalette = {
  contrastText: '#FFF',
  dark: '#e5e5e5',
  light: '#fbfbfb',
  main: '#F7F7F7',
}
export const greenPalette = {
  contrastText: '#FFF',
  dark: '#00ad9d',
  light: '#d4f5f3',
  main: '#00D3CA',
}
export const yellowPalette = {
  contrastText: '#FFF',
  dark: '#e4f45e',
  light: '#f7fccd',
  main: '#EFFC90',
}

export const colorPalettes: IColorPalette = {
  black: blackPalette,
  blue: bluePalette,
  gray: grayPalette,
  green: greenPalette,
  orange: orangePalette,
  primary: redPalette,
  red: redPalette,
  yellow: yellowPalette,
}
