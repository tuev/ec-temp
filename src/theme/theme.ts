export type BreakPoint = {
  xs: '0px'
  sm: '600px'
  md: '960px'
  lg: '1280px'
  xl: '1920px'
}

const breakpoints: BreakPoint = {
  xs: '0px',
  sm: '600px',
  md: '960px',
  lg: '1280px',
  xl: '1920px',
}

export { breakpoints }
