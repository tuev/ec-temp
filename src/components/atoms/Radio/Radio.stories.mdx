import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Radio.stories.tsx'
import Radio from './Radio.tsx'

<Meta title="Radio" component={Radio} />
