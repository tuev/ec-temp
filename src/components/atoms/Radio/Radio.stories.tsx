import React from 'react'
import { action } from '@storybook/addon-actions'
import Radio from './Radio'

export default {
  component: Radio,
  includeStories: ['basic', 'outer'],
  title: 'AppRadio',
}

export const basic = () => (
  <>
    <Radio onChange={action('Radio Clicked!!!')} />
    <Radio onChange={action('Radio Clicked!!!')} checked={true} />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="blue"
      checked={true}
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="green"
      checked={true}
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="gray"
      checked={true}
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="orange"
      checked={true}
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="yellow"
      checked={true}
    />
  </>
)

export const outer = () => (
  <>
    <Radio
      onChange={action('Radio Clicked!!!')}
      variant="outer"
      checked={true}
    />
    <Radio onChange={action('Radio Clicked!!!')} variant="outer" />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="blue"
      variant="outer"
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="green"
      variant="outer"
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="gray"
      variant="outer"
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="orange"
      variant="outer"
    />
    <Radio
      onChange={action('Radio Clicked!!!')}
      customcolor="yellow"
      variant="outer"
    />
  </>
)
