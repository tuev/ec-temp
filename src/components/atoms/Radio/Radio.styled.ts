import { switchProp, prop } from 'styled-tools'
import { space } from 'styled-system'
import { COLOR, colorPalettes } from 'theme/colors'
import { BaseRadioProps } from './Radio.types'
import { createMuiTheme } from '@material-ui/core/styles'
import { Radio } from '@material-ui/core'
import styled, { css } from 'styled-components'
export const AppRadio: React.FC<BaseRadioProps> = styled(Radio)<BaseRadioProps>`
  ${space}
`

export const theme = createMuiTheme({
  palette: {
    primary: {
      contrastText: COLOR.white,
      dark: '#4592f7',
      light: '#beddfe',
      main: COLOR.blue,
    },
  },
})
