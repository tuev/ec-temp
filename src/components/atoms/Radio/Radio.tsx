import React, { FC, useMemo, useCallback } from 'react'
import { AppRadio } from './Radio.styled'
import { BaseRadioProps } from './Radio.types'
import { ThemeProvider, createMuiTheme, Theme } from '@material-ui/core/styles'
import { colorType, colorPalettes } from 'theme/colors'

const getTheme: (color?: colorType) => Theme = (color = 'primary') =>
  createMuiTheme({
    palette: {
      primary: colorPalettes[color],
    },
  })

const Radio: FC<BaseRadioProps> = (props: BaseRadioProps) => {
  const { customColor = 'primary', ...rest } = props
  const theme = getTheme(customColor)

  const getThemeCallback = useCallback(
    (theme: Theme) =>
      createMuiTheme({
        ...theme,
        palette: {
          ...theme.palette,
          primary: colorPalettes[customColor],
        },
      }),
    [customColor]
  )

  return (
    <ThemeProvider theme={theme}>
      <ThemeProvider theme={getThemeCallback}>
        <AppRadio {...rest} color="primary" />
      </ThemeProvider>
    </ThemeProvider>
  )
}

export default Radio
