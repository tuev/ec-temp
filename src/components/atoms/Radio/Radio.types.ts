import { ColorType } from 'theme/colors'
import { SpaceProps } from 'styled-system'
import { RadioProps } from '@material-ui/core/Radio'

import { Theme } from '@material-ui/core/styles''
export interface IRadioProps extends RadioProps {
  customColor?: ColorType
  variant?: 'inner' | 'outer'
}

export type GetTheme = (color?: ColorType) => Theme

export declare type BaseRadioProps = IRadioProps & SpaceProps
