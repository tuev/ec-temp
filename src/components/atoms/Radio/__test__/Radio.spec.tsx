import React from 'react'
import Radio from '..'
import { getTheme } from '../Radio'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { get } from 'lodash'
import { COLOR, ColorType, colorPalettes } from 'theme/colors'
import { getRadioColor } from '../Radio.styled'

describe('Radio', () => {
  it('get radio color', () => {
    const radioColor = getRadioColor({})

    expect(radioColor).toBe(COLOR.primary)
    const colors: ColorType[] = [
      'black',
      'blue',
      'gray',
      'green',
      'orange',
      'primary',
      'red',
      'yellow',
    ]
    colors.forEach(color => {
      const radioCustomColor = getRadioColor({ customcolor: color })
      expect(radioCustomColor).toBe(COLOR[color])
    })
  })

  it('radio theme', () => {
    const customTheme = getTheme()
    const colors: ColorType[] = [
      'primary',
      'blue',
      'black',
      'orange',
      'gray',
      'green',
      'yellow',
      'red',
    ]

    colors.forEach(color =>
      expect(get(getTheme(color), 'palette.primary')).toStrictEqual(
        colorPalettes[color]
      )
    )
    expect(get(customTheme, 'palette.primary.main')).toBe(COLOR.primary)
  })

  it('radio', () => {
    const AppRadio = renderer.create(<Radio />).toJSON()
    expect(AppRadio).toMatchSnapshot()
  })

  it('radio primary', () => {
    const AppRadio = renderer.create(<Radio customcolor="primary" />).toJSON()
    expect(AppRadio).toMatchSnapshot()
  })

  it('radio checked', () => {
    const CheckedAppRadio = renderer.create(<Radio checked={true} />).toJSON()
    expect(CheckedAppRadio).toMatchSnapshot()
  })

  it('radio green checked', () => {
    const CheckedAppRadio = renderer
      .create(<Radio checked={true} customcolor="green" />)
      .toJSON()
    expect(CheckedAppRadio).toMatchSnapshot()
  })
})
