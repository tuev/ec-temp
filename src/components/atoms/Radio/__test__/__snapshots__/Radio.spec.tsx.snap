// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio radio 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root-19 MuiIconButton-root-10 PrivateSwitchBase-root-6 MuiRadio-root-1 MuiRadio-colorPrimary-4 MuiIconButton-colorPrimary-14"
  customcolor="primary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
  variant="inner"
>
  <span
    className="MuiIconButton-label-18"
  >
    <input
      className="PrivateSwitchBase-input-9"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="PrivateRadioButtonIcon-root-22"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-25"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-25 PrivateRadioButtonIcon-layer-23"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </div>
  </span>
</span>
`;

exports[`Radio radio checked 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root-105 MuiIconButton-root-96 PrivateSwitchBase-root-92 MuiRadio-root-87 MuiRadio-colorPrimary-90 PrivateSwitchBase-checked-93 Mui-checked MuiIconButton-colorPrimary-100"
  customcolor="primary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
  variant="inner"
>
  <span
    className="MuiIconButton-label-104"
  >
    <input
      checked={true}
      className="PrivateSwitchBase-input-95"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="PrivateRadioButtonIcon-root-108 PrivateRadioButtonIcon-checked-110"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-111"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-111 PrivateRadioButtonIcon-layer-109"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </div>
  </span>
</span>
`;

exports[`Radio radio green checked 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root-138 MuiIconButton-root-129 PrivateSwitchBase-root-125 MuiRadio-root-120 MuiRadio-colorPrimary-123 PrivateSwitchBase-checked-126 Mui-checked MuiIconButton-colorPrimary-133"
  customcolor="green"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
  variant="inner"
>
  <span
    className="MuiIconButton-label-137"
  >
    <input
      checked={true}
      className="PrivateSwitchBase-input-128"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="PrivateRadioButtonIcon-root-141 PrivateRadioButtonIcon-checked-143"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-144"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-144 PrivateRadioButtonIcon-layer-142"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </div>
  </span>
  <span
    className="MuiTouchRipple-root-163"
  />
</span>
`;

exports[`Radio radio primary 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root-52 MuiIconButton-root-43 PrivateSwitchBase-root-39 MuiRadio-root-34 MuiRadio-colorPrimary-37 MuiIconButton-colorPrimary-47"
  customcolor="primary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
  variant="inner"
>
  <span
    className="MuiIconButton-label-51"
  >
    <input
      className="PrivateSwitchBase-input-42"
      onChange={[Function]}
      type="radio"
    />
    <div
      className="PrivateRadioButtonIcon-root-55"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-58"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-58 PrivateRadioButtonIcon-layer-56"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </div>
  </span>
  <span
    className="MuiTouchRipple-root-77"
  />
</span>
`;
